central/detection/service/service_impl.go

```diff
@@ -219,44 +220,68 @@ func (s *serviceImpl) runDeployTimeDetect(ctx context.Context, enrichmentContext
 	return s.enrichAndDetect(ctx, enrichmentContext, deployment, policyCategories...)
 }
 
-func getObjectsFromYAML(yamlString string) ([]k8sRuntime.Object, error) {
+func getObjectsFromYAML(yamlString string) ([]k8sRuntime.Object, []string, error) {
 	reader := yaml.NewYAMLReader(bufio.NewReader(bytes.NewBufferString(yamlString)))
 	var objects []k8sRuntime.Object
+	var ignoredObjectsRefs []string
 	for {
 		yamlBytes, err := reader.Read()
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
-			return nil, errox.InvalidArgs.New("failed to read yaml").CausedBy(err.Error())
+			return nil, nil, errox.InvalidArgs.New("failed to read yaml").CausedBy(err.Error())
 		}
 		obj, _, err := workloadDeserializer.Decode(yamlBytes, nil, nil)
 		if err != nil {
-			return nil, errox.InvalidArgs.New("could not parse yaml").CausedBy(err.Error())
+			// Only return errors if the resource's schema is not registered.
+			if !k8sRuntime.IsNotRegisteredError(err) {
+				return nil, nil, errox.InvalidArgs.New("could not parse yaml").CausedBy(err.Error())
+			}
+			// Save the ignored object, so we can return it to the caller and skip it.
+			ignoredObj, err := getIgnoredObjectRefFromYAML(string(yamlBytes))
+			if err != nil {
+				return nil, nil, errox.InvariantViolation.New("could not get ignored object").CausedBy(err.Error())
+			}
+			ignoredObjectsRefs = append(ignoredObjectsRefs, ignoredObj)
+			continue
 		}
+
 		if list, ok := obj.(*coreV1.List); ok {
-			listResources, err := getObjectsFromList(list)
+			listResources, ignoredObjs, err := getObjectsFromList(list)
 			if err != nil {
-				return nil, err
+				return nil, nil, err
 			}
 			objects = append(objects, listResources...)
+			ignoredObjectsRefs = append(ignoredObjectsRefs, ignoredObjs...)
 		} else {
 			objects = append(objects, obj)
 		}
 	}
-	return objects, nil
+	return objects, ignoredObjectsRefs, nil
 }
 
-func getObjectsFromList(list *coreV1.List) ([]k8sRuntime.Object, error) {
+func getObjectsFromList(list *coreV1.List) ([]k8sRuntime.Object,
+	[]string, error) {
 	objects := make([]k8sRuntime.Object, 0, len(list.Items))
+	var ignoredObjectsRefs []string
 	for i, item := range list.Items {
 		obj, _, err := workloadDeserializer.Decode(item.Raw, nil, nil)
 		if err != nil {
```

Comment:
if we invert this condition we can reduce indentation level and make this more readable as we will have short circuit when everything is fine and then not indented ignored objects handling.